doc/api/http.md

```diff
@@ -48,26 +48,39 @@ list like the following:
 added: v0.3.4
 -->
 
-The HTTP Agent is used for pooling sockets used in HTTP client
-requests.
-
-The HTTP Agent also defaults client requests to using
-`Connection: keep-alive`. If no pending HTTP requests are waiting on a
-socket to become free the socket is closed. This means that Node.js's
-pool has the benefit of keep-alive when under load but still does not
-require developers to manually close the HTTP clients using
-KeepAlive.
-
-If you opt into using HTTP KeepAlive, you can create an Agent object
-with that flag set to `true`.  (See the [constructor options][].)
-Then, the Agent will keep unused sockets in a pool for later use.  They
-will be explicitly marked so as to not keep the Node.js process running.
-However, it is still a good idea to explicitly [`destroy()`][] KeepAlive
-agents when they are no longer in use, so that the Sockets will be shut
-down.
-
-Sockets are removed from the agent's pool when the socket emits either
-a `'close'` event or a special `'agentRemove'` event. This means that if
+An `Agent` is responsible for managing connection persistence
+and reuse for HTTP clients. It maintains a queue of pending requests
+for a given host and port, reusing a single socket connection for each
+until the queue is empty, at which time the socket is destroyed.
+When a socket connection is no longer writable, for example, if a server
+closes the connection, the socket is destroyed.
+
+Unless an `Agent` is expressly provided in the [`http.request()`]
+options, all HTTP client requests will use the default
+[`http.globalAgent`].
+
+In addition to managing connection persistence for queued HTTP
+requests, an `Agent` can be used for pooling sockets across multiple
+client requests. When providing `{ keepAlive: true }` among the
+[constructor options], the `Agent` will not destroy sockets when
+the request queue has been emptied. Rather, it will pool these
+sockets for later requests to the same host.
+
+By providing `{ keepAlive: true }` as a constructor option to the the
```

Comment:
I think this should be merged with the previous paragraph, otherwise it kind of seems like it's talking about a different situation, when in fact it's just providing more description of `keepAlive: true`.

Also, I think it would be good to note that a socket *will be destroyed* if the server responds with a `Connection: close`.